mne/epochs.py

```diff
@@ -814,13 +814,19 @@ def _reject_setup(self, reject, flat):
                     "%s." % (key.upper(), key.upper())
                 )
 
-        # check for invalid values
-        for rej, kind in zip((reject, flat), ("Rejection", "Flat")):
-            for key, val in rej.items():
-                if val is None or val < 0:
-                    raise ValueError(
-                        '%s value must be a number >= 0, not "%s"' % (kind, val)
-                    )
+            # check for invalid values
+            for rej, kind in zip((reject, flat), ("Rejection", "Flat")):
+                for key, val in rej.items():
+                    name = f"{kind} dict value for {key}"
+                    if callable(val):
+                        continue
+                    _validate_type(val, "numeric", name, extra="or callable")
+                    if val is None or val < 0:
+                        raise ValueError(
+                            """If using numerical %s criteria, the value
+                                must be >= 0 Not '%s'."""
+                            % (kind, val)
+                        )
```

Comment:
Nowadays we prefer f-strings so might as well modernize since the lines are in the diff anyway. Also the triple-quoted string change you made would lead to a ton of spaces in the output error message
```suggestion
                        raise ValueError(
                            f"If using numerical {name} criteria, the value "
                            f"must be >= 0, not {repr(val)}"
                        )
```